### PR TITLE
Fix music get artist artwork bug

### PR DESCRIPTION
**Pull request by DaveTBlake**
When adding artist roles, the query to the song_artist table used when getting artist artwork was overlooked. The results in errors in `CMusicDatabase::GetArtistArtForItem` from GetSingleValue because more than once record may exist with iOrder=0 if artist performing other roles have been scanned from tags.

### CODE REVIEW

**Comment by DaveTBlake** [comment]
jenkins build this please

File: xbmc/music/MusicDatabase.cpp [hunk 1]
```diff
@@ -5954,7 +5954,12 @@ bool CMusicDatabase::GetArtistArtForItem(int mediaId, const std::string &mediaTy
     if (NULL == m_pDB.get()) return false;
     if (NULL == m_pDS2.get()) return false; // using dataset 2 as we're likely called in loops on dataset 1
 
-    std::string sql = PrepareSQL("SELECT type,url FROM art WHERE media_id=(SELECT idArtist from %s_artist WHERE id%s=%i AND iOrder=0) AND media_type='artist'", mediaType.c_str(), mediaType.c_str(), mediaId);
+    std::string sql;
+    if (mediaType == MediaTypeAlbum)
+      sql = PrepareSQL("SELECT type, url FROM art WHERE media_id=(SELECT idArtist from album_artist WHERE idAlbum=%i AND iOrder=0) AND media_type='artist'", mediaId);
+    else
+      sql = PrepareSQL("SELECT type, url FROM art WHERE media_id=(SELECT idArtist from song_artist WHERE idSong=%i AND idRole=1 AND iOrder=0) AND media_type='artist'", mediaId);
```
Comment (Razzeee) [inline xbmc/music/MusicDatabase.cpp]: Can you break these statements into multiple lines so they are better readable and also explain the idRole=1 with an -- comment. Just say what idRole is.

**Comment by DaveTBlake** [comment]
Thanks for caring @Razzeee, look OK to you now? I would like to get this fix into the nightlies soonest, it is a silly omission.

**Comment by Tolriq** [comment]
Is is sure that artist will have the iorder 0 or is the iorder relative to all roles ?

If not then should be order by iOrder limit 1 no ? As currently can return no result.

File: xbmc/music/MusicDatabase.cpp [hunk 2]
```diff
@@ -5954,7 +5954,16 @@ bool CMusicDatabase::GetArtistArtForItem(int mediaId, const std::string &mediaTy
     if (NULL == m_pDB.get()) return false;
     if (NULL == m_pDS2.get()) return false; // using dataset 2 as we're likely called in loops on dataset 1
 
-    std::string sql = PrepareSQL("SELECT type,url FROM art WHERE media_id=(SELECT idArtist from %s_artist WHERE id%s=%i AND iOrder=0) AND media_type='artist'", mediaType.c_str(), mediaType.c_str(), mediaId);
+    std::string sql;
+    if (mediaType == MediaTypeAlbum)
+      sql = PrepareSQL("SELECT type, url FROM art WHERE media_id=(SELECT idArtist from album_artist "
```
Comment (Razzeee) [inline xbmc/music/MusicDatabase.cpp]: You might want to add a LIMIT 1 to the subquery, if you really want to be sure that it doesn't cause a crash. (applies to all four queries)

**Comment by DaveTBlake** [comment]
>Is is sure that artist will have the iorder 0 or is the iorder relative to all roles ?

It is sure that the first artist with role = "artist" will have iOrder = 0. It is not relative to all roles.

The subquery can have no result if there is no artist, and the main query can also have no result if there is no artwork, that is all OK.

As MySQL has issues with LIMIT within subquery, and because this aspect has been working fine so far, I am reluctant to add a LIMIT clause. The only reason we got multiple values was because song_artist now contains values for other roles, and even then it didn't "crash" just logged an error that I could debug. I would rather get the error, and know that something odd has happened in the data than force a single row.

**Comment by Razzeee** [comment]
fine for me. 

jenkins build this please

**Pull request merged**